modules/generator/processor/servicegraphs/store/store.go

```diff
@@ -15,120 +15,103 @@ var _ Store = (*store)(nil)
 
 type store struct {
 	l   *list.List
-	mtx *sync.RWMutex
+	mtx sync.Mutex
 	m   map[string]*list.Element
 
-	evictCallback Callback
-	ttl           time.Duration
-	maxItems      int
+	onComplete Callback
+	onExpired  Callback
+
+	ttl      time.Duration
+	maxItems int
 }
 
-func NewStore(ttl time.Duration, maxItems int, evictCallback Callback) Store {
+// NewStore creates a Store to build service graphs. The store caches edges, each representing a
+// request between two services. Once an edge is complete its metrics can be collected. Edges that
+// have not found their pair are deleted after ttl time.
+func NewStore(ttl time.Duration, maxItems int, onComplete, onExpired Callback) Store {
```

Comment:
Nit: can we unify naming on callbacks? i.e. `onCompleted` or `onExpire`

Reply:
Done :+1: